### DISCUSSION TITLE
CMakeFiles/rtigo3.dir/imgui/imgui_impl_glfw_gl3.cpp.o: undefined reference to symbol 'glEnable'

### DISCUSSION THREAD

**Discussion by bipul-mohanto** [ocornut/imgui#5454] [New Users Build/Link/Run issues ONLY!]
Hi!
On Ubuntu 22.04, I am trying to build a NVIDIA's sample code repository which is using OptiX API (C++). I installed required libraries: `sudo apt-get install -y libglfw3-dev libglew-dev libdevil-dev libassimp-dev libdart-external-imgui-dev libimgui-dev`. 

`cmake ..` command is working fine, but during `make` I am getting this error:

```javascript
/usr/bin/ld: CMakeFiles/rtigo3.dir/imgui/imgui_impl_glfw_gl3.cpp.o: undefined reference to symbol 'glEnable'
/usr/bin/ld: /lib/x86_64-linux-gnu/libGL.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [apps/rtigo3/CMakeFiles/rtigo3.dir/build.make:824: bin/rtigo3] Error 1
make[1]: *** [CMakeFiles/Makefile2:307: apps/rtigo3/CMakeFiles/rtigo3.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```
I am not sure wheather this is an error for `imgui` or not. Could you suggest anything?

**Comment by ocornut** [comment]
You need to link with OpenGL library.

**Reply by bipul-mohanto** [reply]
I am still in building the project phase in cmake. How can I link with OpenGL?

**Reply by ocornut** [reply]
I don’t use cmake or linux but I imagine searching for “ linux cmake link opengl” on the web will give you plenty of answers.

**Reply by bipul-mohanto** [reply]
adding the following lines solved my problem: 

```
include_directories( ${OPENGL_INCLUDE_DIRS}   )

target_link_libraries(${proj_name} ${OPENGL_LIBRARIES} }
```
